addons/l10n_es_edi_tbai/models/account_move.py

```diff
@@ -0,0 +1,246 @@
+    l10n_es_tbai_qr_escaped = fields.Char(string="TicketBAI QR code, escaped", compute="_compute_l10n_es_tbai_qr")
+    l10n_es_tbai_signature = fields.Char(string="TicketBAI XML signature", compute="_compute_l10n_es_tbai_values")
+    l10n_es_tbai_registration_date = fields.Date(
+        string="TicketBAI registration date",
+        help="Technical field to keep the date the invoice was sent the first time as the date the invoice was "
+             "registered into the TicketBai system.",
+        compute="_compute_l10n_es_tbai_values",
+    )
+
+    # Optional fields
+    l10n_es_tbai_refund_reason = fields.Selection(
+        selection=[
+            ('R1', "R1: Art. 80.1, 80.2, 80.6 and rights founded error"),
+            ('R2', "R2: Art. 80.3"),
+            ('R3', "R3: Art. 80.4"),
+            ('R4', "R4: Art. 80 - other"),
+            ('R5', "R5: Factura rectificativa en facturas simplificadas"),
+        ],
+        string="Invoice Refund Reason Code (TicketBai)",
+        help="BOE-A-1992-28740. Ley 37/1992, de 28 de diciembre, del Impuesto sobre el "
+        "Valor Añadido. Artículo 80. Modificación de la base imponible.",
+        copy=False,
+    )
+
+    # -------------------------------------------------------------------------
+    # COMPUTE & API-DECORATED METHODS
+    # -------------------------------------------------------------------------
+
+    @api.depends('move_type', 'company_id')
+    def _compute_l10n_es_tbai_is_required(self):
+        for move in self:
+            move.l10n_es_tbai_is_required = move.is_sale_document() \
+                and move.country_code == 'ES' \
+                and move.company_id.l10n_es_tbai_tax_agency
+
+    def l10n_es_tbai_is_in_chain(self):
+        """
+        True iff invoice has been posted to the chain and confirmed by govt.
+        Note that cancelled invoices remain part of the chain.
+        """
+        tbai_doc_ids = self.edi_document_ids.filtered(lambda d: d.edi_format_id.code == 'es_tbai')
+        return self.l10n_es_tbai_is_required \
+            and len(tbai_doc_ids) > 0 \
+            and not any(tbai_doc_ids.filtered(lambda d: d.state == 'to_send'))
+
+    @api.depends('l10n_es_tbai_is_required', 'name')
+    def _compute_l10n_es_tbai_seq_num(self):
+        tbai_required = self.filtered(lambda m: m.l10n_es_tbai_is_required)
+        for move in tbai_required:
+            sequence, number = move.name.rsplit('/', 1)
+            sequence = regex_sub(r"[^0-9A-Za-z.\_\-\/]", "", sequence)  # remove forbidden characters
+            sequence = regex_sub(r"[\s]+", " ", sequence)  # no more than once consecutive whitespace allowed
+            # NOTE (optional) not recommended to use chars out of ([0123456789ABCDEFGHJKLMNPQRSTUVXYZ.\_\-\/ ])
+            move.l10n_es_tbai_sequence = sequence + ("TEST" if move.company_id.l10n_es_edi_test_env else "")
+            move.l10n_es_tbai_number = regex_sub(r"[^0-9]", "", number)  # remove non-decimal characters
```

Comment:
Leaving a comment open in case we want to enforce all-numbers move names (if that's not the case by default) or any other choice.
For now, just removed the regex_sub.